plugins/GSdx/GSTextureCache.h

```diff
@@ -29,6 +29,82 @@ class GSTextureCache
 public:
 	enum {RenderTarget, DepthStencil};
 
+	template <class T>
+	struct Element {
+		T  data;
+		uint16 next_index;
+		uint16 prev_index;
+	};
+
+	template <class T>
+	class FastList {
+	public:
+		// Seems like 512 is a reasonable upper bound for the size (ZoE2 goes just below 500), maybe 1024 better
+		static const uint16 SIZE = 512;
+		// The last element of the list has next index equal to END
+		static const uint16 END = SIZE;
+		// m_buffer[0] is the head of the list
+		Element<T> m_buffer[SIZE];
```

Comment:
A vector does invalidate all the iterators when the underlying array is resized thus would invalidate the iterators saved in the Source objects during list insertion (for future fast deletion).
So yes I cannot imagine a compact storage with iterators that do not invalidate on list operations that has not fixed size.
Maybe a hierarchy std::list -> m_buffer[SIZE] dynamically managed could work in principle, using a cache friendly SIZE and ensuring that the number or nodes of the std::list are minimized.
Logic would become unfriendly though.